bugtool/cmd/root.go

```diff
@@ -315,6 +319,39 @@ func runAll(commands []string, cmdDir string, k8sPods []string) {
 			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown)
 			continue
 		}
+
+		if strings.Contains(cmd, "xfrm") {
+			var output []string
+			execResult, _ := execCommand(cmd)
+			// splitting the result as an array of string
+			lines := strings.Split(execResult, "\n")
+			// list of substrings to search for in lines
+			substrings := [...]string{"auth", "enc", "aead", "auth-trunc", "comp"}
+			for _, line := range lines {
+				for _, substr := range substrings {
+					// checking if the line contains our substring
+					matched, _ := regexp.MatchString(substr, line)
+					if matched {
+						// looking for hex substring which starts with '0x'
+						for i, splits := range strings.Fields(line) {
+							hexFound, _ := regexp.MatchString("0x", splits)
+							if hexFound {
+								splits := strings.Fields(line)
+								pass := strings.Fields(line)[i]
+								hash := md5.Sum([]byte(pass))
+								splits[i] = "[hash: " + hex.EncodeToString(hash[:]) + "]"
+								line = strings.Join(splits, " ")
+							}
+						}
+					}
+
+				}
+				// after processing appending the line to output
+				output = append(output, line+"\n")
+
+			}
+			log.Print(output)
```

Comment:
Where will the logs be printed?

Re-reading the code in `runAll()`, I think we want to call `writeCmdToFile()` even for the functions with `xfrm`, so that the output remains consistent for all commands (for example we want them to run on the same K8s pods, and we want them all formatted in accordance with `enableMarkdown`). [Apologies if I misled you by telling we should hook in `runAll()` previously.]

What we could do is
1. Extend `writeCmdToFile()` so it takes an additional `filter` callback. If the callback is defined, it is run on the `output` string returned by `execCommand()`, before processing/writing it. For existing commands, run `writeCmdToFile()` with a callback at `nil`.
2. Add your code for hashing the keys as a new, separate function for the file.
3. Pass the callback to commands with `xfrm` in the Go subroutine in `runAll()`, something like:
   ```go
   if strings.Contains(cmd, "xfrm") {
			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown, hashEncryptionKeys)
   } else {
			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown, nil)
   }
   ```

What do you think?

Reply:
ThankYou so much, I was really confused about how to write output to files. This really helped.

Reply:
also i learned about callbacks in go .